test/Microsoft.TestPlatform.AcceptanceTests/RunsettingsTests.cs

```diff
@@ -258,9 +268,156 @@ public void TestAdapterPathFromRunSettings(RunnerInfo runnerInfo)
+                                        </Execution>
+                                    </LegacySettings>
+                                   </RunSettings>";
+            File.WriteAllText(this.runsettingsPath, runsettingsXml);
+            var arguments = PrepareArguments(testAssemblyPath, string.Empty, this.runsettingsPath, this.FrameworkArgValue, runnerInfo.InIsolationValue);
+            arguments = string.Concat(arguments, " /testcasefilter:Name~TimeTest");
+
+            this.InvokeVsTest(arguments);
+
+            this.ValidateSummaryStatus(1, 1, 0);
+        }
+
+        [TestMethod]
+        [NetFullTargetFrameworkDataSource(inIsolation: true, useCoreRunner: false)]
+        public void LegacySettingsAssemblyResolution(RunnerInfo runnerInfo)
+        {
+            AcceptanceTestBase.SetTestEnvironment(this.testEnvironment, runnerInfo);
+            var testAssemblyPath = this.GetAssetFullPath("LegacySettingsUnitTestProject.dll");
+            var runsettingsFormat = @"<RunSettings>
+                                    <MSTest><ForcedLegacyMode>true</ForcedLegacyMode></MSTest>
+                                    <LegacySettings>
+                                        <Execution>
+                                         <TestTypeSpecific>
+                                          <UnitTestRunConfig testTypeId=""13cdc9d9-ddb5-4fa4-a97d-d965ccfc6d4b"">
+                                           <AssemblyResolution>
+                                              <TestDirectory useLoadContext=""true"" />
+                                              <RuntimeResolution>
+                                                  <Directory path=""{0}"" includeSubDirectories=""true"" />
+                                              </RuntimeResolution>
+                                           </AssemblyResolution>
+                                          </UnitTestRunConfig>
+                                         </TestTypeSpecific>
+                                        </Execution>
+                                    </LegacySettings>
+                                   </RunSettings>";
+
+            var testAssemblyDirectory = Path.Combine(this.testEnvironment.TestAssetsPath, "LegacySettingsUnitTestProject", "DependencyAssembly");
+            var runsettingsXml = string.Format(runsettingsFormat, testAssemblyDirectory);
+
+            File.WriteAllText(this.runsettingsPath, runsettingsXml);
+            var arguments = PrepareArguments(testAssemblyPath, string.Empty, this.runsettingsPath, this.FrameworkArgValue, runnerInfo.InIsolationValue);
+            arguments = string.Concat(arguments, " /testcasefilter:Name=DependencyTest");
+
+            this.InvokeVsTest(arguments);
+
+            this.ValidateSummaryStatus(1, 0, 0);
+        }
+
+        #endregion
+
         private string GetRunsettingsFilePath(Dictionary<string, string> runConfigurationDictionary)
         {
             var runsettingsPath = Path.Combine(
```

Comment:
Can we have better way than checking in assembly to source control?

Reply:
This is test scenario, dint want a lot of engineering around this.

Reply:
> This is test scenario, dint want a lot of engineering around this.

Sorry, Couldn't agree on this. We could have move the dependent file(DependencyAssemblyForTest.dll) to required directory during test arrange step(I think this will be few line of code)? 
Following are two reasons I couldn't agree.
- [Don't ever commit binary files to Git](https://robinwinslow.uk/2013/06/11/dont-ever-commit-binary-files-to-git/)
- Readability. Need to use ILSpy to understand `DependencyAssemblyForTest.Class1` .

@singhsarab  Can you please raise another PR to address this? Let me know if you need help. We can set time bound on this.